### PR TITLE
 Improve the exception message when an option within the rootpath section is missing from the user configuration file

### LINKED ISSUE #2235: Improve the exception message when an option within the `rootpath` section is missing from the user configuration file
**Describe the bug**
I ran the `recipe_extreme_events.yml` at the MO for the first time this morning and I got the following error message:

```
KeyError: 'default rootpath must be specified in config-user file'
```

I was confused as I do have a `rootpath` section in my `config-user.yml` file and I run recipes all the time without errors.

The problem was that the `OBS6` option was missing from the `rootpath` section in the `config-user.yml` file. The exception message, unfortunately, didn't help me figure this out.

I'm hoping it will be possible to add the option name to this exception message to make it clear how to fix the problem 😊

### PR DESCRIPTION
**Pull request by ehogan**
## Description

Closes #2235

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/local.py [hunk 1]
```diff
@@ -414,7 +408,8 @@ def _get_rootpath(project):
                     key, ', '.join(str(p) for p in nonexistent))
                 _ROOTPATH_WARNED.add((key, nonexistent))
             return rootpath[key]
-    raise KeyError('default rootpath must be specified in config-user file')
+    raise KeyError(f'The "{project}" option is missing from the "rootpath" '
+                   'section in the config-user.yml file.')
```
Comment (ehogan) [inline esmvalcore/local.py:412]: The pre-commit hooks made all the changes in this PR, except for this one, which is the one I made :)

File: tests/unit/local/test_get_rootpath.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@mock.patch("os.path.exists")
+def test_get_rootpath_does_not_exist(mexists):
+    mexists.return_value = False
+    cfg = {"rootpath": {"CMIP5": ["path1"], "CMIP6": ["path2"]}}
+    project = "OBS"
+    with mock.patch.dict(local.CFG, cfg):
+        msg = rf"The \"{project}\" option is missing.*"
+        with pytest.raises(KeyError, match=msg):
+            local._get_rootpath(project)
```
Comment (valeriupredoi) [inline tests/unit/local/test_get_rootpath.py:28]: well, you made these changes too, unless precommit hooks now write unit tests too (which would be fab) :grin:

**Review by valeriupredoi — APPROVED** [review]
cheers @ehogan = looking good and very useful!

File: tests/unit/local/test_get_rootpath.py (hunk 2 above)
Comment (ehogan) [reply tests/unit/local/test_get_rootpath.py:28]: 🤣 🤣 🤣 I added these after I made that comment! 😝

**Comment by valeriupredoi** [comment]
cheers @ehogan :beer: @zklaus I reckon this can go in 2.10 but you da big honcho :cowboy_hat_face:

**Pull request merged**